### PR TITLE
Add Julia doc to emphasize compile_model args in StanModel; re issue #64

### PR DESCRIPTION
**Pull request by roualdes**
Re Issue #64, duplicate some doc to emphasize that `StanModel` can accept and pass to `compile_model` the arguments to tweak compilation, `make_args`, and Stanc transpilation `stanc_args`.

### CODE REVIEW

File: julia/src/model.jl [hunk 1]
```diff
@@ -6,15 +6,19 @@ mutable struct StanModelStruct end
 
 A StanModel instance encapsulates a Stan model instantiated with data.
```
Comment (bob-carpenter) [inline julia/src/model.jl:7]: Is this the standard way to do Julia doc, with a line for the class doc and then a line for the constructor?  It looks very strange.

File: julia/src/model.jl [hunk 2]
```diff
@@ -6,15 +6,19 @@ mutable struct StanModelStruct end
...
     StanModel(;stan_file, data="", seed=204, chain_id=0)
 
 Construct a StanModel instance from a `.stan` file, compiling if necessary.
 
-This is equivalent to calling `compile_model` and then the original constructor of StanModel.
+    StanModel(;stan_file, stanc_args=[], make_args=[], data="", seed=204, chain_id=0)
+
+Construct a StanModel instance from a `.stan` file, compiling if
+necessary.  This is equivalent to calling `compile_model` and then the
+original constructor of StanModel.
```
Comment (bob-carpenter) [inline julia/src/model.jl:21]: StanModel -> `StanModel`

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
I'm not a Julia dev, so please take this with a grain of salt.  The only comment I'm confident about is the one about objects vs. string literals.

File: julia/src/model.jl (hunk 1 above)
Comment (roualdes) [reply julia/src/model.jl:7]: As far as I know and see.  Agreed, it looks strange.

File: julia/src/model.jl [hunk 3]
```diff
@@ -6,15 +6,19 @@ mutable struct StanModelStruct end
 
 A StanModel instance encapsulates a Stan model instantiated with data.
 
-The constructor a Stan model from the supplied library file path and data. Data
+Construct a Stan model from the supplied library file path and data. Data
 should either be a string containing a JSON object or a path to a data file ending in `.json`.
```
[the remark this answers is not in the document]
Comment (roualdes) [reply julia/src/model.jl]: Ripgrep suggests we have 17 other occurrences of this typo. I'll submit a different PR for the others.

[the remark this answers is not in the document]
Comment (bob-carpenter) [reply julia/src/model.jl]: Sounds good.

**Comment by bob-carpenter** [comment]
@roualdes---there were still a couple residual comments.  Did you just want me to go ahead and merge anyway?  I don't think we have to be super-formal about this---I was just supplying feedback and am happy to merge anything that makes the doc better.

File: julia/src/model.jl [hunk 4]
```diff
@@ -6,15 +6,19 @@ mutable struct StanModelStruct end
...
 
     StanModel(;stan_file, data="", seed=204, chain_id=0)
 
 Construct a StanModel instance from a `.stan` file, compiling if necessary.
 
-This is equivalent to calling `compile_model` and then the original constructor of StanModel.
+    StanModel(;stan_file, stanc_args=[], make_args=[], data="", seed=204, chain_id=0)
+
+Construct a StanModel instance from a `.stan` file, compiling if
+necessary.  This is equivalent to calling `compile_model` and then the
```
[the remark this answers is not in the document]
Comment (roualdes) [reply julia/src/model.jl]: Attempted a rewrite.  Let me know what you think, or just merge if it is satisfactory.

File: julia/src/model.jl (hunk 2 above)
Comment (roualdes) [reply julia/src/model.jl]: Yep, missed a few of these.  Fixed now.

**Comment by roualdes** [comment]
Hopefully I addressed the residual comments here.

**Review by bob-carpenter — APPROVED** [review]

**Pull request merged**